### PR TITLE
dist/tools/suit: convert to package

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

Fetch the suit-manifest-generator from upstream instead of keeping a copy in the RIOT repository.

### Testing procedure

All SUIT examples should still compile, manifests should be generated.

### Issues/PRs references

depends on https://github.com/RIOT-OS/riotdocker/pull/114, https://github.com/RIOT-OS/riotdocker/pull/118

### CODE REVIEW

**Comment by aabadie** [comment]
needs rebase

**Comment by benpicco** [comment]
I guess we could just patch the package to remove `pyhsslms`, we don't use it.
That way we don't have to wait for all the workers to get updated again.

File: dist/tools/suit/Makefile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+PKG_NAME=suit-manifest-generator
+PKG_URL=https://github.com/ARMmbed/suit-manifest-generator.git
+PKG_VERSION=959dd3fca00806faf06699c449653cd48b84841e
+PKG_LICENSE=Apache2.0
+
+include $(RIOTBASE)/pkg/pkg.mk
+
+all:
+	$(Q)rm -fr suit-manifest-generator
+	-$(Q)ln -s $(PKGDIRBASE)/suit-manifest-generator .
```
Comment (aabadie) [inline dist/tools/suit/Makefile:10]: This looks ugly. You could either change the value of `SUIT_TOOL` defined in

https://github.com/RIOT-OS/RIOT/blob/453d0478dce107f810f4690093899e36faaf7cb8/makefiles/suit.base.inc.mk#L2-L3

to `$(PKGDIRBASE)/suit-manifest-generator/bin/suit-tool`

Or specify `PKG_SOURCE_DIR` before the include of `pkg.mk`:

```mk
PKG_SOURCE_DIR ?= $(CURDIR)
include $(RIOTBASE)/pkg/pkg.mk
```
Both options are untested but I think they are better than the current symlink approach.

In any case:
```suggestion
	:
```

**Review by aabadie — CHANGES_REQUESTED** [review]
I have a few comments. The build system integration can be improved.

**Comment by aabadie** [comment]
Needs a rebase

**Comment by benpicco** [comment]
Unfortunately `tests/suit_manifest` still fails with

```
--- testing manifest 0
Unable to parse COSE signature
---- res=-1 (expected=-6)
```

maybe @bergzand still knows how our version of this tool differs from upstream?

**Comment by fjmolinas** [comment]
Does this still fail?

**Comment by benpicco** [comment]
I don't see why it would not, upstream didn't change.

**Pull request closed without merging**